vms/platformvm/camino_service.go

```diff
@@ -63,6 +85,135 @@ type GetConfigurationReply struct {
+	}
+
+	unlockedOutputs := map[ids.ID]uint64{}
+	bondedOutputs := map[ids.ID]uint64{}
+	depositedOutputs := map[ids.ID]uint64{}
+	depositedBondedOutputs := map[ids.ID]uint64{}
+
+utxoFor:
+	for _, utxo := range utxos {
+		assetID := utxo.AssetID()
+		switch out := utxo.Out.(type) {
+		case *secp256k1fx.TransferOutput:
+			newBalance, err := math.Add64(unlockedOutputs[assetID], out.Amount())
+			if err != nil {
+				unlockedOutputs[assetID] = stdmath.MaxUint64
+			} else {
+				unlockedOutputs[assetID] = newBalance
+			}
+		case *locked.Out:
+			switch out.LockState() {
+			case locked.StateBonded:
+				newBalance, err := math.Add64(bondedOutputs[assetID], out.Amount())
+				if err != nil {
+					bondedOutputs[assetID] = stdmath.MaxUint64
+				} else {
+					bondedOutputs[assetID] = newBalance
+				}
+			case locked.StateDeposited:
+				newBalance, err := math.Add64(depositedOutputs[assetID], out.Amount())
+				if err != nil {
+					depositedOutputs[assetID] = stdmath.MaxUint64
+				} else {
+					depositedOutputs[assetID] = newBalance
+				}
+			case locked.StateDepositedBonded:
+				newBalance, err := math.Add64(depositedBondedOutputs[assetID], out.Amount())
+				if err != nil {
+					depositedBondedOutputs[assetID] = stdmath.MaxUint64
+				} else {
+					depositedBondedOutputs[assetID] = newBalance
+				}
+			default:
+				service.vm.ctx.Log.Warn("Unexpected utxo lock state")
+				continue utxoFor
+			}
+		default:
+			service.vm.ctx.Log.Warn("unexpected output type in UTXO",
+				zap.String("type", fmt.Sprintf("%T", out)),
+			)
+			continue utxoFor
+		}
+
+		response.UTXOIDs = append(response.UTXOIDs, &utxo.UTXOID)
+	}
+
+	balances := map[ids.ID]uint64{}
+	for assetID, amount := range unlockedOutputs {
+		newBalance, err := math.Add64(balances[assetID], amount)
+		if err != nil {
+			balances[assetID] = stdmath.MaxUint64
+		} else {
+			balances[assetID] = newBalance
+		}
+	}
+	for assetID, amount := range bondedOutputs {
+		newBalance, err := math.Add64(balances[assetID], amount)
+		if err != nil {
+			balances[assetID] = stdmath.MaxUint64
+		} else {
+			balances[assetID] = newBalance
+		}
+	}
+	for assetID, amount := range depositedOutputs {
+		newBalance, err := math.Add64(balances[assetID], amount)
+		if err != nil {
+			balances[assetID] = stdmath.MaxUint64
+		} else {
+			balances[assetID] = newBalance
+		}
+	}
+	for assetID, amount := range depositedBondedOutputs {
+		newBalance, err := math.Add64(balances[assetID], amount)
+		if err != nil {
+			balances[assetID] = stdmath.MaxUint64
+		} else {
+			balances[assetID] = newBalance
+		}
+	}
+
+	response.Balances = newJSONBalanceMap(balances)
+	response.Unlockeds = newJSONBalanceMap(unlockedOutputs)
+	response.BondedOutputs = newJSONBalanceMap(bondedOutputs)
+	response.DepositedOutputs = newJSONBalanceMap(depositedOutputs)
+	response.DepositedBondedOutputs = newJSONBalanceMap(depositedBondedOutputs)
+	response.Balance = response.Balances[service.vm.ctx.AVAXAssetID]
```

Comment:
We should think about omitting the following fields, theya re duplicate.

Reply:
single json values (Bonded, Deposited, DepositedBonded) removed. Balance belongs to the original struct